### PR TITLE
Fits reader fix

### PR DESCRIPTION
**Pull request by bramalingam**
Files for testing can be downloaded from : 
http://www.euhou.net/index.php/salsaj-software-mainmenu-9/download-mainmenu-10

(Please download the "Version with macros and associated images"
and use the associated images folder, to check if those images open in ImageJ.
All images with "*.fts", "*.FTS" and "*.fits" can be checked with this PR.

@melissalinkert @ximenesuk @jburel

### CODE REVIEW

**Comment by sbesson** [comment]
@bramalingam: can the sample files be copied under the inbox (and eventually I assume we'll want to add some configuration files for automated testing)?

**Comment by melissalinkert** [comment]
Note this will change the MD5s for our existing FITS datasets, so a configuration PR will be necessary no matter what.

**Comment by sbesson** [comment]
Thanks. Moving to breaking so that the FITS datasets configuration PR can be opened and tested without affecting the merge jobs.

File: components/formats-bsd/src/loci/formats/in/FitsReader.java [hunk 1]
```diff
@@ -137,19 +147,29 @@ protected void initFile(String id) throws FormatException, IOException {
...
     while (in.read() == 0x20);
     pixelOffset = in.getFilePointer() - 1;
 
     m.sizeC = 1;
     m.sizeT = 1;
-    if (getSizeZ() == 0) m.sizeZ = 1;
+    if (getSizeZ() == 0) {
+        m.sizeZ = 1;
+    }
+    else m.sizeZ = getSizeZ();
```
Comment (melissalinkert) [inline components/formats-bsd/src/loci/formats/in/FitsReader.java]: I'd expect ```m.sizeZ = getSizeZ()``` to be a no-op - was there a reason this change was necessary?

File: components/formats-bsd/src/loci/formats/in/FitsReader.java [hunk 2]
```diff
@@ -137,19 +147,29 @@ protected void initFile(String id) throws FormatException, IOException {
...
+        m.sizeY = 1;
+        m.sizeZ = 1;
+    }
 
     // correct for truncated files
     int planeSize =
-      getSizeX() * getSizeY() * FormatTools.getBytesPerPixel(getPixelType());
-    if (DataTools.safeMultiply64(planeSize, getSizeZ()) >
+      m.sizeX * m.sizeY * FormatTools.getBytesPerPixel(getPixelType());
+    if (DataTools.safeMultiply64(planeSize, m.sizeZ) >
```
Comment (melissalinkert) [inline components/formats-bsd/src/loci/formats/in/FitsReader.java]: ```m.sizeX``` and ```getSizeX()``` should be equivalent (ditto for Y and Z) - was there a reason for this change?

**Comment by sbesson** [comment]
Closing. Please reopen when the fixes above are pushed.

**Pull request closed without merging**